nixos/modules/services/networking/bind.nix

```diff
@@ -8,32 +8,40 @@ let
 
   bindUser = "named";
 
-  bindZoneOptions = {
-    name = mkOption {
-      type = types.str;
-      description = "Name of the zone.";
-    };
-    master = mkOption {
-      description = "Master=false means slave server";
-      type = types.bool;
-    };
-    file = mkOption {
-      type = types.either types.str types.path;
-      description = "Zone file resource records contain columns of data, separated by whitespace, that define the record.";
-    };
-    masters = mkOption {
-      type = types.listOf types.str;
-      description = "List of servers for inclusion in stub and secondary zones.";
-    };
-    slaves = mkOption {
-      type = types.listOf types.str;
-      description = "Addresses who may request zone transfers.";
-      default = [];
+  bindZoneCoerce = list: builtins.listToAttrs (lib.forEach list (zone: {name = zone.name; value = zone; }));
+
+  bindZoneOptions = { name, config, ... }: {
+    options = {
+      name = mkOption {
+        type = types.str;
+        description = "Name of the zone.";
+      };
+      master = mkOption {
+        description = "Master=false means slave server";
+        type = types.bool;
+      };
+      file = mkOption {
+        type = types.either types.str types.path;
+        description = "Zone file resource records contain columns of data, separated by whitespace, that define the record.";
+      };
+      masters = mkOption {
+        type = types.listOf types.str;
+        description = "List of servers for inclusion in stub and secondary zones.";
+      };
+      slaves = mkOption {
+        type = types.listOf types.str;
+        description = "Addresses who may request zone transfers.";
+        default = [];
+      };
+      extraConfig = mkOption {
+        type = types.str;
+        description = "Extra zone config to be appended at the end of the zone section.";
+        default = "";
+      };
     };
-    extraConfig = mkOption {
-      type = types.str;
-      description = "Extra zone config to be appended at the end of the zone section.";
-      default = "";
+
+    config = {
+      name = mkDefault name;
```

Comment:
Is there any reason to allow this to be overridden *without* the use of mkForce?

Reply:
If the user want to use the key as a name to identify something, instead of the actual name, then mkForce would prohibit that usecase. Ie. I might call my zone for "mymainzone" which uses `name = "eyjhb.dk"` :)

Reply:
`mkForce` should only be required for overriding if it has the potential to have unintended or breaking side effects. For example, when you need to configure a service to output to systemd instead of a log file we should do that without a `mkDefault` like
```nix
{
  settings.logTo = "systemd";
}
```

such that users are required to use `mkForce` when overriding. In this case this breaks the assumption that service logs can be found in journald.

For things like default settings or names, using `mkDefault` is appropriate to allow overriding without `mkForce`.

Reply:
Ended up keeping as-is, chat here - https://logs.nix.samueldr.com/nixos-dev/2021-05-03#1620048631-1620045059;